simpleviewinc/keg-herkin
# Iframe and bugfix

**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-440)

## Context

* front-end update to the test reporter

## Goal

* utilize the backend test reporter api and display it in the runner tab UI
* Update the reporter API to return a json result if it's called from an API and return the html/txt if its called directly from the browser url
* update the runner tab to show the test reporter if it exists

## Updates
* `repos/backend/endpoints/reports.js`
  * update so it returns a valid json if we call it from the app
* `repos/tap/assets/icons/externalLink.js`
  * new icon for opening in a new tab
* `repos/tap/components/labels/prefixTitleHeader.js`
  * extracted from the `Surface` component to be reused in a different component
* `repos/tap/containers/screens/runnerScreen.js`
  * update so we can see the report in an iframe
* `repos/tap/utils/api/apiRequest.js`
  * `devLog` returns a falsy so we would never get that empty array
  * updated so it just returns false if the apiRequest fails     

## Testing

* pull this branch
* run `keg herkin start --config ./repos/example/herkin.config.js --from keg-herkin:iframe-and-bugfix`
* navigate to http://herkin-iframe-and-bugfix.local.kegdev.xyz/
* navigate to the `Runner` tab
* verify that it shows an empty screen saying `No Report to show`
* on your terminal, run `keg herkin cr test`
  * ensure that it is successful
  * [image](https://user-images.githubusercontent.com/3317835/110190103-e0a42e80-7dde-11eb-9223-9fe7725d07a8.png)
 
* either refresh the browser or re-render the `Test Runner` screen by switching to a different tab and then back
* Verify that you now see the test report
  * [image](https://user-images.githubusercontent.com/3317835/110190074-cb2f0480-7dde-11eb-891e-3d4e819b3993.png)
